File: JavaScript/Problem002.js
const solution = (num) => {
  let fib1 = 1, fib2 = 2
  let sum = 2
  for (let i = fib1; fib1 < num; ++i) {
    let fib = fib1 + fib2
    fib1 = fib2
    fib2 = fib
    if (fib % 2 == 0) {
      sum += fib
    }
  }
  return sum
}

console.log(solution(4000000))

File: Go/Problem002.go
package main 

import "fmt"

func main() {
    sum, i := 0, 1
    for fibonacci(i) < 4e6 {
        if (fibonacci(i) % 2 == 0) {
            sum += fibonacci(i);
        }
        i++
    }
    fmt.Println(sum)
}

func fibonacci(num int) int {
    fib1, fib2 := 1, 1
    for i := 1; i < num; i++ {
        fib := fib1 + fib2
        fib1 = fib2
        fib2 = fib
    }
    return fib1
}